### PR TITLE
[Ed2Nav] Remove backup nav file

### PR DESCRIPTION
**Pull request by nonifier**
If we have successfully written and renamed the new data.nav file, we don't need the old one any more.

If you still want to retrieve the previous file, there is a daily backup done by the DevOps team that can be recovered from on demand.

### CODE REVIEW

File: source/ed/tests/ed2nav_test.cpp [hunk 1]
```diff
@@ -72,11 +72,9 @@ BOOST_AUTO_TEST_CASE(writing_test_ed2nav_should_not_throw) {
     // Checks if the already existing binary is renamed and the new one is created
     BOOST_CHECK(ed::write_data_to_file(test, data));
     BOOST_CHECK(boost::filesystem::exists("ed2nav_test.nav.lz4"));
-    BOOST_CHECK(boost::filesystem::exists("ed2nav_test.nav.lz4.bak"));
```
Comment (AurelienLP) [inline source/ed/tests/ed2nav_test.cpp:75]: Why not BOOST_CHECK(!boost::filesystem::exists("ed2nav_test.nav.lz4.bak")); instead ?

File: source/ed/ed2nav.cpp [hunk 2]
```diff
@@ -261,6 +261,18 @@ bool rename_file(const std::string& source_name, const std::string& dest_name) {
     return true;
 }
 
+bool remove_file(const std::string& filename) {
+    if(!boost::filesystem::exists(filename))
```
Comment (AurelienLP) [inline source/ed/ed2nav.cpp:265]: if (..) {
 // blabla
}

**Review by AurelienLP — APPROVED** [review]

File: source/ed/ed2nav.cpp (hunk 2 above)
Comment (nonifier) [reply source/ed/ed2nav.cpp:265]: Do you really need this ?
```cpp
// if the filename doesn't exist, it's already deleted....
```

Comment (AurelienLP) [reply source/ed/ed2nav.cpp:265]: I mean, you need to put the braces after the if, even if you have one instruction after :stuck_out_tongue:

Comment (nonifier) [reply source/ed/ed2nav.cpp:265]: oooh, I see!
 
It's weird, I thought `clang-format` would do that type of formatting automatically :thinking:
I think we should rather harden the clang-formatting config to enforce this behaviour.

Comment (AurelienLP) [reply source/ed/ed2nav.cpp:265]: I thought it would too, or maybe it is clang-tidy... :thinking:

**Review by benoit-bst — APPROVED** [review]

**Review by mbounabi — APPROVED** [review]

**Review by pbench — APPROVED** [review]

File: source/ed/ed2nav.cpp (hunk 2 above)
Comment (nonifier) [reply source/ed/ed2nav.cpp:265]: yup, looks like it's a clang-tidy thing - https://clang.llvm.org/extra/clang-tidy/checks/readability-braces-around-statements.html

**Pull request merged**